### PR TITLE
feat/Implemented user login 

### PR DESCRIPTION
**Pull request by poneciak57**
#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2611; New code is covered with tests and coverage requirements are met.
&#x2611; Build is not failing.
&#x2611; All tests are passing.
&#x2611; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2611; All issue requirements are implemented.
&#x2610; At least two reviewers are assigned.
&#x2611; The Pull request is linked to the related issue.
&#x2610; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2611; Correct unit test name - suffix 'Test' in the end.
&#x2611; Correct integration test name - suffix 'IT' in the end.
<s>&#x2610; Positive and negative test paths have been implemented.</s>

### CODE REVIEW

**Review by dawciobiel — APPROVED** [review]
Looks good to me

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes

**Review by dawciobiel — DISMISSED** [review]
LGTM

**Review by dawciobiel — DISMISSED** [review]
LGTM

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/UserRegistrationIT.java [hunk 1]
```diff
@@ -5,21 +5,21 @@
...
 import pl.byczazagroda.trackexpensesappbackend.repository.UserRepository;
-import org.springframework.test.web.servlet.MockMvc;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
+@Import(WebSecurityConfig.class)
```
Comment (mirekgab) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/UserRegistrationIT.java:22]: In an integration test, you don't need to import the configuration explicitly.

**Review by mirekgab — CHANGES_REQUESTED** [review]

**Review by Iwona007 — CHANGES_REQUESTED** [review]
Request changes

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/UserLoginIT.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(controllers = AuthController.class,
+        includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes =
+                {UserServiceImpl.class, ErrorStrategy.class, WebSecurityConfig.class}))
+@ActiveProfiles("test")
+public class UserLoginIT {
+
```
Comment (Iwona007) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/UserLoginIT.java:36]: Why you don't extends BaseIntegrationTestIT?

**Review by Iwona007 — CHANGES_REQUESTED** [review]
Request changes

**Review by mirekgab — DISMISSED** [review]

**Review by Iwona007 — APPROVED** [review]

**Review by mirekgab — APPROVED** [review]

**Pull request merged**